src/systemctl/systemctl-util.c

```diff
@@ -880,26 +880,27 @@ bool output_show_unit(const UnitInfo *u, char **patterns) {
         return true;
 }
 
-bool install_client_side(void) {
-        /* Decides when to execute enable/disable/... operations client-side rather than server-side. */
+InstallClientSide install_client_side(void) {
+        /* Decides whether to execute enable/disable/… client-side offline operation rather than
+         * server-side. */
 
         if (running_in_chroot_or_offline())
-                return true;
+                return INSTALL_CLIENT_SIDE_OFFLINE;
 
         if (sd_booted() <= 0)
-                return true;
+                return INSTALL_CLIENT_SIDE_NOT_BOOTED;
 
         if (!isempty(arg_root))
-                return true;
+                return INSTALL_CLIENT_SIDE_ARG_ROOT;
 
         if (arg_runtime_scope == RUNTIME_SCOPE_GLOBAL)
-                return true;
+                return INSTALL_CLIENT_SIDE_GLOBAL_SCOPE;
 
         /* Unsupported environment variable, mostly for debugging purposes */
         if (getenv_bool("SYSTEMCTL_INSTALL_CLIENT_SIDE") > 0)
-                return true;
+                return INSTALL_CLIENT_SIDE_OVERRIDE;
 
```

Comment:
If you touch this, I'd like to move the argument/envvar checks to the top so that we don't needlessly check the state of local system.

Reply:
Hmm, not resolved completely? The arg_root check can be moved too. If operating on a chroot whether the host root is booted or not is irrelevant